core/src/test/scala/quasar/destination/gbq/GBQDestinationModuleSpec.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.destination.gbq
+
+import argonaut._, Argonaut._
+
+import cats.effect.{IO, Timer, ContextShift}
+
+import quasar.EffectfulQSpec
+import quasar.api.destination.DestinationError
+import quasar.api.destination.DestinationType
+import quasar.connector.destination.Destination
+import quasar.api.destination.DestinationError.InitializationError
+import quasar.connector.ResourceError
+import quasar.contrib.scalaz.MonadError_
+
+import java.nio.file.{Files, Paths}
+import java.nio.charset.StandardCharsets.UTF_8
+
+import scala.{
+  Either,
+  Left,
+  Right
+}
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scala.Predef.String
+
+object GBQDestinationModuleSpec extends EffectfulQSpec[IO] {
+  val authCfgPath = Paths.get(getClass.getClassLoader.getResource("precog-ci-275718-e913743ebfeb.json").toURI)
+  val authCfgString = new String(Files.readAllBytes(authCfgPath), UTF_8)
+  val authCfgJson: Json = Parse.parse(authCfgString) match {
+    case Left(value) => Json.obj("malformed" := true)
+    case Right(value) => value
+  }
+
+  "initialization" should {
+    "fail with malformed config when config is not decodable" >>* {
+      val cfg = Json("malformed" := true)
+      dest(cfg)(r => IO { r match {
+        case Left(DestinationError.MalformedConfiguration(_, c, _)) =>
+          c must_=== jString(GBQDestinationModule.Redacted)
+        case _ => ko("Expected a malformed configuration")
```

Comment:
And maybe this should be `Left(MalformedConfiguration`, right?

Reply:
Here I'm expecting only a failure, a `DestinationError.MalformedConfiguration` any thing else should fail the test. So just used `ko()` in order to make sure I return a `MatchResult`.

Reply:
`must beLike { ... }` solves this problem by accepting a partial function, which obviates the need for fall-through case.